CHAOS-5623 Release notes 1.38.1 (#6857)
harness/developer-hub: release-notes/chaos-engineering.md

## May 2024

### Version 1.38.1

#### New features and enhancements

- This release optimizes the experiment flow by:
    - Reading environment variables from the chaos engine.
    - Eliminating the experiment's custom resources and the corresponding steps for new experiments.
    - Eliminating the **install experiment** step.
    - Reducing the length of the YAML manifest.
    - Increasing the speed of execution of the experiment.
    - Adding all the overrides to the chaos engine.

>     - Enhancing the filter that lists the infrastructure that is compatible only with the new experiment template. (CHAOS-5122)

Reviewer: ```suggestion
    - Enhancing the infrastructure list filter, compatible only with the new experiment template. (CHAOS-5122)(CHAOS-5122)
```

:::info note
The existing APIs will work as per the norm on old and new chaos infrastructure, whereas new experiments will work only on the updated infrastructure (infrastructure version >= 1.38.0).
:::

#### Fixed issues

- Fixed an issue where the compatibility check was enabled for other infrastructure types too. The overview form now preserves the state while switching between different infrastructures. (CHAOS-5614)

- Fixed the rendering of the **View Onboarding Progress** page. (CHAOS-5583)

- Fixed an issue where the user could not set up or create a Datadog probe. (CHAOS-5440)

- Fixed an issue where the [pod IO stress](/docs/chaos-engineering/chaos-faults/kubernetes/pod/pod-io-stress) experiment incorrectly applied stress on the helper pod instead of the target container. (CHAOS-5416)

> - Fixed an issue where a user could not connect to a ChaosHub if the secret associated with it had a '-' symbol (after the deployment of ng-manager 1.33). (CHAOS-5112)

Reviewer: ```suggestion
- Fixed an issue where a user could not connect to a ChaosHub if its secret had a '-' symbol (after the deployment of ng-manager 1.33). (CHAOS-5112)
```
